File: chap19_retina/Peskin_lateral_inhibition.m

% The Matlab/Octave file used to generate the image in Fig. 15 of Chapter 19
% Solutions of the Peskin lateral inhibition  model 
% Keener and Sneyd, Mathematical Physiology, 3rd Edition, Springer.

% Written by James Keener and James Sneyd

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
clear all
close all
clc
set(0,                           ...
   'defaultaxesfontsize', 20,   ...
   'defaultaxeslinewidth', 2.0, ...
   'defaultlinelinewidth', 2.0, ...
   'defaultpatchlinewidth', 0.7);

%parameters
lam = 1;
k = 20;

t = [0:.01:3];

R = 1/(lam+1) -(k-1)/(k-lam-1)*exp(-k*t) + lam*k/((k-lam-1)*(lam+1))*exp(-(lam+1)*t);  

x = [-2:.01:2];
E = (x>=0);
sql = sqrt(lam+1);
Ix=lam/(lam+1)*((x>=0).*(1-exp(-x*sql)/2) +(x<0).*exp(x*sql)/2);


figure(1)
plot(t,R)
box off
xlabel('t')
ylabel('R(t)')

figure(2)
plot(x,E,'--',x,E-Ix)
box off
legend('boxoff')
legend('stimulus, E','response, R','location','northwest')
xlabel('x')

File: chap19_retina/Peskin_lateral_inhibition.py

#     -------------------------------------------------------------------
# 
#      Solutions of the Peskin lateral inhibition  model.
# 
#      For Chapter 19, Section 19.4.1 of
#      Keener and Sneyd, Mathematical Physiology, 3rd Edition, Springer.
# 
#      Written by James Keener and James Sneyd.
# 
#     -------------------------------------------------------------------

import numpy as np
import matplotlib.pyplot as plt

lam = 1
k = 20

t = np.arange(0, 3, 0.01)
R = 1/(lam + 1) - (k - 1)/(k - lam - 1) * np.exp(-k * t) + lam * k / ((k - lam - 1) * (lam + 1)) * np.exp(-(lam + 1) * t)

x = np.arange(-2, 2, 0.01)
E = (x >= 0)
sql = np.sqrt(lam + 1)
Ix = lam / (lam + 1) * ((x >= 0) * (1 - np.exp(-x * sql) / 2) + (x < 0) * np.exp(x * sql) / 2)

plt.figure()
plt.plot(t, R)
plt.xlabel('t')
plt.ylabel('R(t)')

plt.figure()
plt.plot(x, E, '--', label='stimulus, E')
plt.plot(x, E - Ix, label='response, R')
plt.xlabel('x')
plt.legend()



